apollo-router/src/plugins/telemetry/metrics/apollo.rs

```diff
@@ -153,45 +152,66 @@ impl ApolloMetricsExporter {
 
         // This is the thread that actually sends metrics
         tokio::spawn(async move {
-            // We want to collect stats into batches, but also send periodically if a batch is not filled.
-            // This implementation is not ideal as we do have to store all the data when really it could be folded as it is generated.
-            // But in the interested of getting something over the line quickly let's go with this as it is simple to understand.
-            rx.chunks_timeout(DEFAULT_BATCH_SIZE, Duration::from_secs(10))
-                .for_each(|reports| async {
-                    let aggregated_report = Report::new(reports);
-
-                    match pool.get().await {
-                        Ok(mut reporter) => {
-                            let report = aggregated_report.into_report(header.clone());
-                            match reporter
-                                .submit(apollo_spaceport::ReporterRequest {
-                                    apollo_key: apollo_key.clone(),
-                                    report: Some(report),
-                                })
-                                .await
-                            {
-                                Ok(_) => {}
-                                Err(e) => {
-                                    tracing::warn!("failed to submit stats to spaceport: {}", e);
-                                }
-                            };
+            let timeout = tokio::time::interval(Duration::from_secs(5));
+            let mut report = Report::default();
+            tokio::pin!(timeout);
+
+            loop {
+                tokio::select! {
+                    single_report = rx.next() => {
+                        if let Some(r) = single_report {
+                            report += r;
+                        } else {
+                            break;
                         }
-                        Err(err) => {
-                            tracing::warn!(
-                                "stats discarded as unable to get connection to spaceport: {}",
-                                err
-                            );
+                       },
+                    _ = timeout.tick() => {
+                        if report.operation_count != 0 {
+                            Self::send_report(&pool, &apollo_key, &header, std::mem::take(&mut report)).await;
                         }
-                    };
-                })
-                .await;
+                    }
+                };
+            }
+
+            Self::send_report(&pool, &apollo_key, &header, report).await;
```

Comment:
This may also send an empty report. Maybe move the check for operation count into send_report

Reply:
good catch, fixed in [40ff7c7](https://github.com/apollographql/router/pull/1066/commits/40ff7c7add681e62a4e9e746178d99e16e364170)